lifebit-ai/h3agwas
# Updates test profile, adds better test

## Description

[h3agwas](https://github.com/h3abionet/h3agwas) is a very customizable gwas pipeline that allows to use many different tools to do the association testing. Problem is that none of those are included by default and I missed that when creating first test profile for this repo.

I added all optional parameters to trigger all optional processes that I could successfully make run offline with test data. I succeeded with four association test, while other 4 failed due to different reasons and will need further debugging. Here is the list of items added and not added to the updated test profile:
- [x] plink association test
- [ ] plink Fisher test
- [ ] plink linear regression test
- [x] plink logistic regression test
- [x] Gemma
- [x] BoltLMM
- [ ] FastLMM
- [ ] FastGWA

Detailed explanation why some of them didn't work is available here:
https://docs.google.com/document/d/1Ac0wVRf2bcDcodLExFQez5EEw1gUiBIMAANPl5bOXc0/edit#heading=h.6qdlytsekren

Old test profile has been renamed to `smalltest`.
New test profile can be run with:
```
nextflow run . -profile test
```
Which is equivalent to
```
nextflow run . -profile smalltest \
    --assoc 1 \
    --adjust \ 
    --logistic 1 \ 
    --gemma 1 \
    --gemma_num_cores 2 \
    --gemma_mem_req 4.GB \
    --boltlmm 1 \
    --bolt_num_cores 2 \
    --bolt_mem_req 4.GB \
    --bolt_covariates_type '0,1' \
    --other_process_memory 4.GB
```


## Testing

To test the new profiles run:
```
git clone  https://github.com/lifebit-ai/h3agwas
cd h3agwas
git checkout vlad-fix-assoc
nextflow run . -profile smalltest
nextflow run . -profile testtest
```
Then check the results folder, especially the final pdf report `out-report.pdf`.

### Local runs
```
nextflow run . -profile smalltest
```
![image](https://user-images.githubusercontent.com/64809705/128872136-7ef4ea67-5861-4618-aada-d16caf5aecc5.png)
```
nextflow run . -profile test
```
![image](https://user-images.githubusercontent.com/64809705/128872104-5c2e62f4-64d5-48a1-a452-067b52a26f93.png)

Plot example from output report:
![image](https://user-images.githubusercontent.com/64809705/128872330-9bc3a15d-2e9a-4041-8977-985df0dca1e8.png)
